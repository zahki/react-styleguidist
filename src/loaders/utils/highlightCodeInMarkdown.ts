import remark from 'remark';
import visit from 'unist-util-visit';
import highlightCode from './highlightCode';

function highlight() {
	return (ast: any) => {
		visit(ast, 'code', (node: any) => {
			node.value = highlightCode(node.value, node.lang);
		});
	};
}

/**
 * Highlight code in code snippets in Markdown.
 *
 * @param {string} markdown
 * @returns {string}
 */
export default function highlightCodeInMarkdown(markdown: string): string {
	return remark().use(highlight).processSync(markdown).toString();
}
