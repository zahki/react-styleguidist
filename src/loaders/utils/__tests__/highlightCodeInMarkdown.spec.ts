import highlightCodeInMarkdown from '../highlightCodeInMarkdown';

it('should highlight code with specified language', () => {
	const text = `
The only true button.

\`\`\`html
<p>Hello React</p>
\`\`\`
`;
	const actual = highlightCodeInMarkdown(text);
	expect(actual).toMatchSnapshot();
});

it('should not highlight code without language', () => {
	const text = `
The only \`true\` button.

\`\`\`
<p>Hello React</p>
\`\`\`
`;
	const actual = highlightCodeInMarkdown(text);
	expect(actual).toMatchSnapshot();
});
