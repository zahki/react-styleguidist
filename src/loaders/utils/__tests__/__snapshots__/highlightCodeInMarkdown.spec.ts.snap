// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should highlight code with specified language 1`] = `
"The only true button.

\`\`\`html
<span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&lt;</span>p</span><span class=\\"token punctuation\\">></span></span>Hello React<span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&lt;/</span>p</span><span class=\\"token punctuation\\">></span></span>
\`\`\`
"
`;

exports[`should not highlight code without language 1`] = `
"The only \`true\` button.

    <p>Hello React</p>
"
`;
